tests/model/test_model.py

```diff
@@ -1360,6 +1361,200 @@ def test_can_user_edit_topic(
         else:
             report_error.assert_called_once_with(expected_response[user_type][0])
 
+    @pytest.mark.parametrize(
+        "topic_name, latest_message_timestamp, server_feature_level,"
+        " topic_editing_limit_seconds, move_messages_within_stream_limit_seconds,"
+        " expected_new_topic_name",
+        [
+            case(
+                "hi!",
+                11662271397,
+                0,
+                None,
+                None,
+                RESOLVED_TOPIC_PREFIX + "hi!",
+                id="topic_resolved:Zulip2.1+:ZFL0",
+            ),
+            case(
+                RESOLVED_TOPIC_PREFIX + "hi!",
+                11662271397,
+                0,
+                None,
+                None,
+                "hi!",
+                id="topic_unresolved:Zulip2.1+:ZFL0",
+            ),
```

Comment:
As mentioned in another comment, if we wish to fully test resolve/unresolve for each of these cases, the resolve/unresolve part could be a nested parametrize, since they are independent.

So you end up with a matrix of parameters from something like the equivalent of
```python3
@pytest.mark.parametrize("param1, param2", ...)
@pytest.mark.parametrize("a, b, c, d", ...)
def test_something(...)